### ISSUE TITLE
转化数据为英文的操作

### ISSUE THREAD

**Issue by damao666** [Determined22/zh-NER-TF#77]

**Comment by coranholmes** [comment]
`vocab_build`方法你有改过吗？
```
            elif ('\u0041' <= word <='\u005a') or ('\u0061' <= word <='\u007a'):
                word = '<ENG>'
```
这一段会把所有英文单词替换成`<ENG>`，这部分代码要做相应修改。

**Comment by damao666** [comment]
感谢回复，您好，我是一名学生，现在需要用这个方法做一个对照实验，用来写论文，比较急，但是数据来源是英文的，希望得到您的帮助，我愿意提供100元的酬劳，如果愿意的话请加我的QQ851958123，非常感谢。

**Comment by damao666** [comment]
> `vocab_build`方法你有改过吗？
> 
> ```
>             elif ('\u0041' <= word <='\u005a') or ('\u0061' <= word <='\u007a'):
>                 word = '<ENG>'
> ```
> 这一段会把所有英文单词替换成`<ENG>`，这部分代码要做相应修改。

感谢回复，您好，我是一名学生，现在需要用这个方法做一个对照实验，用来写论文，比较急，但是数据来源是英文的，希望得到您的帮助，我愿意提供100元的酬劳，如果愿意的话请加我的QQ851958123，非常感谢。

**Comment by damao666** [comment]
> `vocab_build`方法你有改过吗？
> 
> ```
>             elif ('\u0041' <= word <='\u005a') or ('\u0061' <= word <='\u007a'):
>                 word = '<ENG>'
> ```
> 这一段会把所有英文单词替换成`<ENG>`，这部分代码要做相应修改。

您好，谢谢回复，我是一名学生，现在需要这个方法做对照实验，写论文，但是我用的是英文数据，希望得到您的帮助，愿意提供100元的酬劳，如果愿意的话请加我QQ851958123，非常感谢。

**Comment by coranholmes** [comment]
可以参考一下我fork的版本，改成了英文可用的，支持自定义标签，python 2.7.5下跑通。https://github.com/coranholmes/zh-NER-TF

**Comment by damao666** [comment]
> 可以参考一下我fork的版本，改成了英文可用的，支持自定义标签，python 2.7.5下跑通。https://github.com/coranholmes/zh-N

我把vocab_build 改了后可以用了，但是准确率很低，我在挑挑参数，非常感谢回复

**Comment by damao666** [comment]
> `vocab_build`方法你有改过吗？
> 
> ```
>             elif ('\u0041' <= word <='\u005a') or ('\u0061' <= word <='\u007a'):
>                 word = '<ENG>'
> ```
> 这一段会把所有英文单词替换成`<ENG>`，这部分代码要做相应修改。
我把这个删了，跑出来了结果，最后准确率不到1%。是我数据集不好吗，心好累。。


(tensorflow-gpu) D:\主动学习\bilstm+crf对照实验\zh-NER-TF-master>python main.py --mode=test --demo_model=1550825705
vocab_size: 8607
Namespace(CRF=True, batch_size=64, clip=5.0, demo_model='1550825705', dropout=0.5, embedding_dim=300, epoch=40, hidden_dim=300, lr=0.001, mode='test', optimizer='Adam', pretrain_embedding='random', shuffle=True, test_data='data_path', train_data='data_path', update_embedding=True)
.\data_path_save\1550825705\checkpoints/model-3760
D:\software\Anaconda\aa\envs\tensorflow-gpu\lib\site-packages\tensorflow\python\ops\gradients_impl.py:112: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "
test data: 3958
=========== testing ===========
INFO:tensorflow:Restoring parameters from .\data_path_save\1550825705\checkpoints/model-3760
Restoring parameters from .\data_path_save\1550825705\checkpoints/model-3760
processed 120668 tokens with 1593 phrases; found: 147 phrases; correct: 1.
accuracy:  98.56%; precision:   0.68%; recall:   0.06%; FB1:   0.11
PER: precision:   0.68%; recall:   0.06%; FB1:   0.11  147

**Comment by coranholmes** [comment]
我好像就用的默认参数没怎么调就用了原作者默认的，除了输入格式以外模型我好像也没怎么改，效果还行啊……你把错误的那些数据打出来看看错的是哪些？不知道你用的标注数据集质量如何？用的是有名的标注数据集还是自己的数据呢？如果是自己的数据集可能会有一些bias。

**Comment by damao666** [comment]
> 我好像就用的默认参数没怎么调就用了原作者默认的，除了输入格式以外模型我好像也没怎么改，效果还行啊……你把错误的那些数据打出来看看错的是哪些？不知道你用的标注数据集质量如何？用的是有名的标注数据集还是自己的数据呢？如果是自己的数据集可能会有一些bias。

非常感谢您回复我的问题。谢谢您！
我用的是自己的数据集，但是crf跑出来还40%的准确率。
我只识别一类 实体，而且都是单个单词，我就把data.py 中的改成以下类型。
## tags, BIO
tag2label = {"O": 0,
             "B-PER": 1, "I-PER": 2
             }

还有train.data中的数据格式如下所示（因为是单个词，所以全部标注为B-PER）：
...
than O
antiparkinsonian O
drug O
rasagiline B-PER
and O
selegiline B-PER
consider O
...

还有 def vocab_build(vocab_path, corpus_path, min_count): 方法中注释掉以下内容
A-Z：(\u0041-\u005a)    a-z ：\u0061-\u007a
            # elif ('\u0041' <= word <='\u005a') or ('\u0061' <= word <='\u007a'):
            #     word = '<ENG>'

其他部分都没改，结果不好，是不是我数据集有问题呢。

**Comment by coranholmes** [comment]
我的猜测是因为B-PER这个tag在数据中整体占比偏小，大部分都是O，所以模型倾向于给数据标注O，所以你的recall极低，这样可能需要更改一下loss函数，我记得issue里有人问过这个问题，作者有回复，你看看吧。

**Comment by damao666** [comment]
> 我的猜测是因为B-PER这个tag在数据中整体占比偏小，大部分都是O，所以模型倾向于给数据标注O，所以你的recall极低，这样可能需要更改一下loss函数，我记得issue里有人问过这个问题，作者有回复，你看看吧。

好的， 非常感谢你，我试一下。

**Comment by damao666** [comment]
> 我的猜测是因为B-PER这个tag在数据中整体占比偏小，大部分都是O，所以模型倾向于给数据标注O，所以你的recall极低，这样可能需要更改一下loss函数，我记得issue里有人问过这个问题，作者有回复，你看看吧。


我的结果准确率还行，召回率太低了，有的连1%不到。不知道为什么，心累。

**Comment by handsomeTeng** [comment]
> 
> 
> 我的猜测是因为B-PER这个tag在数据中整体占比偏小，大部分都是O，所以模型倾向于给数据标注O，所以你的recall极低，这样可能需要更改一下loss函数，我记得issue里有人问过这个问题，作者有回复，你看看吧。

您好，方便留一下联系方式么，我也有几个问题想请教一下您，我的q761768935
